src/editor/components/modals/GeoModal/GeoModal.component.jsx

```diff
@@ -0,0 +1,151 @@
+import { useState, useCallback, useEffect } from 'react';
+
+import styles from './GeoModal.module.scss';
+import { Copy32Icon, Mangnifier20Icon } from '../../../icons';
+
+import { useAuthContext } from '../../../contexts/index.js';
+import Modal from '../Modal.jsx';
+import { Button, Input } from '../../components/index.js';
+import { GoogleMap, useJsApiLoader, Marker } from '@react-google-maps/api';
+import { DownloadIcon } from '../../../icons/icons.jsx';
+import GeoImg from '../../../../../ui_assets/geo.png';
+
+const GeoModal = ({ isOpen, onClose }) => {
+  const { currentUser } = useAuthContext();
+  const isProUser = currentUser && currentUser.isBeta;
+  const [markerPosition, setMarkerPosition] = useState({
+    lat: 0,
+    lng: 0,
+    elevation: 0
+  });
+
+  const roundToSix = (num) => {
+    return Math.round(num * 1e6) / 1e6;
+  };
+
+  useEffect(() => {
+    // get coordinate data in this format: {latitude: ..., longitude: ..., elevation: ...}
+    const coord = AFRAME.scenes[0].getAttribute('metadata')['coord'];
+    if (coord) {
+      const lat = roundToSix(parseFloat(coord.latitude));
+      const lng = roundToSix(parseFloat(coord.longitude));
+      const elevation = parseFloat(coord.elevation) || 0;
+
+      if (!isNaN(lat) && !isNaN(lng)) {
+        setMarkerPosition({ lat, lng, elevation });
+      }
+    }
+  }, []);
+
+  const onMapClick = useCallback((event) => {
+    setMarkerPosition((prev) => ({
+      ...prev,
+      lat: roundToSix(event.latLng.lat()),
+      lng: roundToSix(event.latLng.lng())
+    }));
+  }, []);
+
+  const { isLoaded } = useJsApiLoader({
+    googleMapsApiKey: 'AIzaSyCwldpJKOZ1yh_FT8NrUPuPeMedEy1R2jw'
```

Comment:
The key should be in .env.development and .env.production.

Reply:
but this will open the key, it can be viewed through dev tools. Or is it ok, @kfarr ?